api/api.go

```diff
@@ -0,0 +1,118 @@
+package api
+
+import (
+	"errors"
+	"io"
+	"syscall"
+	"time"
+
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+var (
+	ErrNotImplemented = errors.New("command not implemented")
+	ErrEmptyID        = errors.New("container id cannot be empty")
+)
+
+type API interface {
```

Comment:
Should we call API as ContainerOperations?

Reply:
Ya, we can work on naming and how big this interface should be for general OCI compatibility.

Reply:
What do you think about adding `context.Context` to these methods?  It's not really important with the implementation side, but if we want to build clients the context is useful for `exec.CommandContext` to support timeouts and such.